src/core/tsi/alts/zero_copy_frame_protector/alts_grpc_integrity_only_record_protocol.cc

```diff
@@ -42,34 +42,33 @@ typedef struct alts_grpc_integrity_only_record_protocol {
 static tsi_result alts_grpc_integrity_only_extra_copy_protect(
     alts_grpc_record_protocol* rp, grpc_slice_buffer* unprotected_slices,
     grpc_slice_buffer* protected_slices) {
-  /* Allocates memory for protected frame.  */
+  /* Allocates memory for protected frame and copies data.  */
   size_t protected_frame_size =
       unprotected_slices->length + rp->header_length + rp->tag_length;
   grpc_slice protected_slice = GRPC_SLICE_MALLOC(protected_frame_size);
+  uint8_t* data = GRPC_SLICE_START_PTR(protected_slice) + rp->header_length;
+  for (size_t i = 0; i < unprotected_slices->count; i++) {
+    memcpy(data, GRPC_SLICE_START_PTR(unprotected_slices->slices[i]),
+           GRPC_SLICE_LENGTH(unprotected_slices->slices[i]));
+    data += GRPC_SLICE_LENGTH(unprotected_slices->slices[i]);
+  }
   /* Calls alts_iovec_record_protocol protect.  */
   char* error_details = nullptr;
   iovec_t header_iovec = {GRPC_SLICE_START_PTR(protected_slice),
                           rp->header_length};
   iovec_t tag_iovec = {GRPC_SLICE_START_PTR(protected_slice) +
                            rp->header_length + unprotected_slices->length,
                        rp->tag_length};
-  alts_grpc_record_protocol_convert_slice_buffer_to_iovec(rp,
-                                                          unprotected_slices);
+  rp->iovec_buf[0].iov_base =
+      GRPC_SLICE_START_PTR(protected_slice) + rp->header_length;
+  rp->iovec_buf[0].iov_len = unprotected_slices->length;
```

Comment:
Same as above.

Reply:
As above, we only use `unprotected_slices->length` for keeping track of frame data length.